src/main/kotlin/org/rust/ide/intentions/WrapLambdaExprIntention.kt

```diff
@@ -32,9 +35,11 @@ class WrapLambdaExprIntention : PsiElementBaseIntentionAction() {
     override fun invoke(project: Project, editor: Editor, element: PsiElement) {
         val ctx = findContext(element) ?: return
         val lambdaBody = ctx.lambdaExpr.expr ?: return
-        val bodyStr = ctx.lambdaExpr.expr?.text ?: return
+        var bodyStr = ctx.lambdaExpr.expr?.text ?: return
+        bodyStr = "\n$bodyStr\n"
         val blockExpr = RustElementFactory.createBlockExpr(project, bodyStr) ?: return
-        lambdaBody.replace(blockExpr)
+        val offset = ((lambdaBody.replace(blockExpr)) as RustBlockExprElement).block?.expr?.textOffset ?: return
+        editor.caretModel.moveToOffset(offset)
```

Comment:
Yes, that would be less confusing perhaps. @ujpv let's then just leave the caret where it was!